neural_compressor/experimental/compression/pruning.py

```diff
@@ -18,167 +18,276 @@
 
 from neural_compressor.compression.pruner.utils import process_config, parse_to_prune
 from neural_compressor.compression.pruner.pruners import get_pruner
+from neural_compressor.experimental.compression.utils import collect_layer_inputs, get_layers
 from neural_compressor.compression.pruner.utils import logger, torch
+from typing import Optional, Callable
```

Comment:
help move this file to neural_compressor/compression/pruner, the experimental folder will be deleted in the future